sdk/lib/rtl/amd64/unwind.c

```diff
@@ -451,44 +590,73 @@ RtlVirtualUnwind(
                 break;
 
             case UWOP_SET_FPREG:
+                Reg = UnwindInfo->FrameRegister;
+                Context->Rsp = GetReg(Context, Reg) - UnwindInfo->FrameOffset * 16;
                 i++;
                 break;
 
             case UWOP_SAVE_NONVOL:
+                Reg = UnwindCode.OpInfo;
+                Offset = *(USHORT*)(&UnwindInfo->UnwindCode[i + 1]);
+                SetRegFromStackValue(Context, ContextPointers, Reg, (DWORD64*)Context->Rsp + Offset);
                 i += 2;
                 break;
 
             case UWOP_SAVE_NONVOL_FAR:
+                Reg = UnwindCode.OpInfo;
+                Offset = *(ULONG*)(&UnwindInfo->UnwindCode[i + 1]);
+                SetRegFromStackValue(Context, ContextPointers, Reg, (DWORD64*)Context->Rsp + Offset);
                 i += 3;
                 break;
 
-            case UWOP_SAVE_XMM:
-                i += 2;
+            case UWOP_EPILOG:
+                i += 1;
                 break;
 
-            case UWOP_SAVE_XMM_FAR:
-                i += 3;
+            case UWOP_SPARE_CODE:
+                __debugbreak();
+                i += 2;
                 break;
 
             case UWOP_SAVE_XMM128:
+                Reg = UnwindCode.OpInfo;
+                Offset = *(USHORT*)(&UnwindInfo->UnwindCode[i + 1]);
+                SetXmmRegFromStackValue(Context, ContextPointers, Reg, (M128A*)(Context->Rsp + Offset));
                 i += 2;
                 break;
 
             case UWOP_SAVE_XMM128_FAR:
+                Reg = UnwindCode.OpInfo;
+                Offset = *(ULONG*)(&UnwindInfo->UnwindCode[i + 1]);
+                SetXmmRegFromStackValue(Context, ContextPointers, Reg, (M128A*)(Context->Rsp + Offset));
                 i += 3;
                 break;
 
             case UWOP_PUSH_MACHFRAME:
-                i += 1;
-                break;
+                Offset = UnwindCode.OpInfo * sizeof(DWORD64);
+                Context->Rip = *(PDWORD64)(Context->Rsp + Offset);
+                Context->Rsp = *(PDWORD64)(Context->Rsp + Offset + 24);
```

Comment:
"24": magic value?

Reply:
yes, sorry :) I improved the code it should be more obvious now.